src/components/FormTable.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { setId } from '../redux/actions/id';
+import { setName } from '../redux/actions/name';
+import { setFatherName } from '../redux/actions/fatherName';
+import { setAge } from '../redux/actions/age';
+import { setAddress } from '../redux/actions/address';
+import { setData } from '../redux/actions/data';
+import { setEditButton } from '../redux/actions/editbutton';
+
+const FormTable = ({
+  data,
+  setId,
+  setName,
+  setFatherName,
+  setAge,
+  setAddress,
+  setEditButton,
+}) => {
+  const handleEdit = (detId) => {
+    setEditButton(true);
+    const tempObj = data.filter((value) => detId === value.id)[0];
```

Comment:
Check always for null & undefined before mapping, filtering or accessing an index of an array